share/models/identifiers.py

```diff
@@ -0,0 +1,57 @@
+from furl import furl
+
+from django.db import models
+
+from share.models.base import ShareObject
+from share.models.fields import ShareForeignKey, ShareURLField
+
+__all__ = ('CreativeWorkIdentifier', 'PersonIdentifier')
+
+
+# TODO Common interface, so we're not duplicating code. Doesn't work because
+# ShareObjectMeta doesn't look at bases when building Version classes.
+#
+# class Identifier(ShareObject):
+#    # http://twitter.com/berniethoughts/, mailto://contact@cos.io
+#    uri = ShareURLField(unique=True)
+#
+#    # twitter.com, cos.io
+#    host = models.TextField(editable=False)
+#
+#    # http, mailto
+#    scheme = models.TextField(editable=False)
+#
+#    def save(self, *args, **kwargs):
+#        f = furl(self.uri)
+#        self.host = f.host
+#        self.scheme = f.scheme
+#        super(Identifier, self).save(*args, **kwargs)
+#
+#    class Meta:
+#        abstract = True
+
+
+class CreativeWorkIdentifier(ShareObject):
+    uri = ShareURLField(unique=True)
+    host = models.TextField(editable=False)
+    scheme = models.TextField(editable=False)
+    creative_work = ShareForeignKey('AbstractCreativeWork', related_name='%(class)ss')
+
+    def save(self, *args, **kwargs):
+        f = furl(self.uri)
```

Comment:
```
>>> f = furl('mailto:foo@bar.com')
>>> f.scheme, f.host
('mailto', 'bar.com')
>>> f = furl('urn://ISSN/1234-1234')
>>> f.scheme, f.host
('urn', 'issn')
```
I fully believe there are other cases it won't handle, but I haven't found them yet. I think it'll be fine, especially with that very nice set of shiny new IRI-parsing links.